### PR TITLE
feat(j-s): Select subpoena type

### PR DESCRIPTION
**Pull request by oddsson**
# Select subpoena type

[Asana](https://app.asana.com/0/322488613152836/1207244440406695/f)

## What

Select either `ARREST` or `ABSENCE` subpoena types. 

## Why

This is a part of the indictments flow we are creating.

## Screenshots / Gifs

image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced a new subpoena functionality with multiple types (e.g., absence, arrest) in the judicial system app.
  - Added a required court room field in court arrangements when necessary.
  
- **Bug Fixes**
  - Changed `policeCaseNumber` prop to `elementId` in indictment case files list.

- **Improvements**
  - Enhanced PDF generation service to include subpoenas.
  - Updated role and permission checks for accessing subpoena PDFs.

- **Styling**
  - Added responsive grid layout specific to subpoena pages.

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Court/Indictments/Subpoena/Subpoena.tsx [hunk 1]
```diff
@@ -92,18 +122,97 @@ const Subpoena: React.FC<React.PropsWithChildren<unknown>> = () => {
...
+                title={`Fyrirkall - ${defendant.name} - PDF`}
+                pdfType="subpoena"
+                disabled={
+                  !courtDate?.date ||
+                  !courtDate?.location ||
+                  !defendant.subpoenaType
+                }
+              />
+            </Box>
+          ))}
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Court/Indictments/Subpoena/Subpoena.tsx]: Good idea, will do.

File: apps/judicial-system/backend/src/app/modules/case/pdf.service.ts [hunk 2]
```diff
@@ -285,4 +287,23 @@
 
     return await this.throttle
   }
+
+  async getSubpoenaPdf(
+    theCase: Case,
+    defendant: Defendant,
+    arraignmentDate?: Date,
+    location?: string,
+    subpoenaType?: SubpoenaType,
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/pdf.service.ts:297]: Will do.

File: apps/judicial-system/web/src/routes/Court/Indictments/Subpoena/Subpoena.tsx [hunk 3]
```diff
@@ -92,18 +122,99 @@ const Subpoena: React.FC<React.PropsWithChildren<unknown>> = () => {
...
+            courtRoomRequired
           />
         </Box>
+        <Box component="section" marginBottom={10}>
+          {workingCase.defendants?.map((defendant, index) => (
+            <Box
+              marginBottom={
+                index + 1 === workingCase.defendants?.length ? 0 : 2
+              }
+            >
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Court/Indictments/Subpoena/Subpoena.tsx]: Done

File: apps/judicial-system/web/src/components/PdfButton/PdfButton.tsx [hunk 4]
```diff
@@ -30,16 +33,16 @@ const PdfButton: React.FC<React.PropsWithChildren<Props>> = ({
...
   const { limitedAccess } = useContext(UserContext)
 
   const handlePdfClick = async () => {
     const prefix = limitedAccess ? 'limitedAccess/' : ''
-    const postfix = pdfType?.endsWith('caseFilesRecord')
-      ? `/${policeCaseNumber}`
-      : ''
-    const url = `${api.apiUrl}/api/case/${caseId}/${prefix}${pdfType}${postfix}`
+    const postfix = elementId ? `/${elementId}` : ''
+    const query = queryParameters ? `?${queryParameters}` : ''
```
Comment (unakb) [inline apps/judicial-system/web/src/components/PdfButton/PdfButton.tsx:44]: Nice 👍

**Review by unakb — APPROVED** [review]

**Pull request merged**